src/ws-week-picker/ws-week-picker-calendar.js

```diff
@@ -0,0 +1,236 @@
+    selectedYear: React.PropTypes.number,
+    selectedWeek: React.PropTypes.number,
+    onChange: React.PropTypes.func
+  };
+
+  /**
+   * @param {Object} props React properties
+   * @constructor
+   */
+  constructor(props) {
+    super(props);
+
+    // if there is a week selected show the according year on start
+    // else show the current year
+    const selectedDate = props.selectedYear != null && props.selectedWeek != null
+      ? getDateOfISOWeek(props.selectedWeek, props.selectedYear) : new Date(Date.now());
+    this.state = {
+      showingYear: selectedDate.getFullYear()
+    };
+
+    const today = new Date(Date.now());
+    this.todayYear = today.getFullYear();
+    this.todayWeek = getWeekOfYear(today);
+  }
+
+  /**
+   * Show the previous year.
+   * @returns {void}
+  */
+  prevYear() {
+    this.setState({
+      showingYear: this.state.showingYear - 1
+    });
+  }
+
+  /**
+   * Show the next year.
+   * @returns {void}
+  */
+  nextYear() {
+    this.setState({
+      showingYear: this.state.showingYear + 1
+    });
+  }
+
+  /**
+   * Checks if a week is selected and therefor equals the input properties.
+   * @param {number} year Year of the week
+   * @param {number} week Week
+   * @returns {boolean}
+  */
+  isActive(year, week) {
+    return this.props.selectedYear === year && this.props.selectedWeek === week;
+  }
+
+  /**
+   * Checks if a week is the current week.
+   * @param {number} year Year of the week
+   * @param {number} week Week
+   * @returns {boolean}
+  */
+  isToday(year, week) {
+    return this.todayYear === year && this.todayWeek === week;
+  }
+
+  /**
+   * Builds an array of rows for the calendar. Every row holds one or none week of the month referenced by the column.
+   * @param {{week: number, year: number}[][]} weeksPerMonth Array that holds 4 or 5 weeks for every column index of allMonth.
+   * @returns {JSX.Elements[]}
+  */
+  buildWeekRows(weeksPerMonth) {
+    // there are up to 5 weeks per month
+    return [0, 1, 2, 3, 4].map(weekIndex =>
+      <tr key={weekIndex}>
+        {
+          allMonths.map((month, monthIndex) => {
+            const weekInMonth = weeksPerMonth[monthIndex][weekIndex];
+            if (weekInMonth == null) return <td key={`${monthIndex}_${weekIndex}`}></td>;
+            const {week, year} = weekInMonth;
+            return (
+              <td className={(monthIndex < 2 || monthIndex > 13 ? 'off ' : '')
+                  + (this.isActive(year, week) ? 'active ' : '')
+                  + (this.isToday(year, week) ? 'today ' : '')}
+                key={`${monthIndex}_${weekIndex}`}
+                onClick={() => this.props.onChange({week, year})}
+              >
+                <a className="week">{week}</a>
+              </td>
+            );
+          })
+        }
+      </tr>);
+  }
+
+  /**
+   * Renders the calendar.
+   * @returns {void}
+   */
+  render() {
+    const weeksPerMonth = [];
+    for (let i = -2; i <= 13; i++) {
```

Comment:
Why do you start here from -2? A comment with an explanation why u choosed the numbers would be cool. Like getting +- 2 month more for better overview.